app/javascript/scripts/new_user_timezone.js

```diff
@@ -0,0 +1,16 @@
+import jstz from 'jstz';
+import setCookie from './set_cookie';
+
+export default () => {
+  const oldIntl = window.Intl;
```

Comment:
Usually, in a situation like this I'd probs chuck this line in the `try/catch` to, just to make things nice and clean.

While it's good to have only the minimal amount of code needed in a `try/catch`, in this situation you've clearly got an alternative you can use for if `Intl` isn't playing nice, and the only time this line should fail would be when the system is drastically compromised (either by 'hacking' or the whole system just crashing as the entire JS engine breaks down).